pkg/cloudevents/transport/http/options.go

```diff
@@ -56,6 +56,28 @@ func WithMethod(method string) Option {
 	}
 }
 
+// WithHeader sets an additional default outbound header for all cloudevents
+// when using an HTTP request.
+func WithHeader(key, value string) Option {
```

Comment:
WithHeader makes it possible to setup a header to be sent for the life of the client but not per Send. Add WithHeader to context as well.